https.go

```diff
@@ -0,0 +1,47 @@
+package dns
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+const mime = "application/dns-udpwireformat"
+
+// MsgToRequest wraps m in a http POST request according to the DNS over HTTPS Spec.
+func MsgToRequest(m *Msg, url string) (*http.Request, error) {
+	out, err := m.Pack()
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest("POST", url, bytes.NewReader(out))
+	req.Header.Set("content-type", mime)
```

Comment:
This will cause an allocation. If you use Go’s header case, that’s avoided. So: Content-Type.